src/zarr/core/array.py

```diff
@@ -547,7 +555,7 @@ async def _create(
         *,
         # v2 and v3
         shape: ShapeLike,
-        dtype: npt.DTypeLike,
+        dtype: npt.DTypeLike | DTypeWrapper[Any, Any],
```

Comment:
It would be nice if this part of the API only accepted one of these types (either native or a Zarr Dtype instance).

Reply:
agreed, I can see how painful it is to push the parsing up